float32 x_relative
float32 y_relative
float32 z_relative